File: Series_4/series4_code/1d-linFEM/plot_fem.py
import numpy, pylab
u = numpy.loadtxt("u_wo_bc_fem.txt")
x = numpy.loadtxt("x_wo_bc_fem.txt")
pylab.plot(x, u, '-o', label='$u(x)$')
pylab.plot(x, numpy.sin(x), label='$\\sin(x)$')
pylab.legend()
pylab.show()

ubc = numpy.loadtxt("u_w_bc_fem.txt")
xbc = numpy.loadtxt("x_w_bc_fem.txt")
pylab.plot(xbc, ubc, '-o', label='$u(x)$')
pylab.plot(xbc, numpy.cos(xbc)+ 3./(4*numpy.pi)*xbc+3.0/4.0, label='$\\cos(x)+\\frac{3}{4\\pi}x + \\frac{3}{4}$')
pylab.legend()
pylab.show()

File: Series_4/series4_code/1d-linFEM/plot_fem.m
u = load("u_wo_bc_fem.txt");
x = load("x_wo_bc_fem.txt");
plot(x, u, '-o');
hold on
plot(x, sin(x))
leg = legend('$u(x)$', '$\\sin(x)$');
set(leg, 'Interpreter', 'latex');
set(leg, 'FontSize', 12);
xlabel("x");
ylabel("u");
title("Without BC");
disp('Press any key to continue...')
pause
cla

ubc = load("u_w_bc_fem.txt");
xbc = load("x_w_bc_fem.txt");
plot(xbc, ubc, '-o');
hold on
plot(xbc, cos(xbc)+ 3./(4*pi)*xbc+3.0/4.0);
leg = legend('$u(x)$', '$\\cos(x)+\frac{3}{4\pi}x + \frac{3}{4}$');
set(leg, 'Interpreter', 'latex');
set(leg, 'FontSize', 12);
xlabel("x");
ylabel("u");
title("Without BC");
